core/lib/spree/testing_support/preferences.rb

```diff
@@ -29,6 +29,33 @@ def assert_preference_unset(preference)
         find("#preferences_#{preference}")['checked'].should be false
         Spree::Config[preference].should be false
       end
+
+      def stub_spree_preferences(preferences)
+        preferences.each do |name, value|
+          if Spree::Config.method(:[]).owner >= Spree::Config.class
+            allow(Spree::Config).to receive(:[]).and_call_original
+          end
+          allow(Spree::Config).to receive(:[]).with(name) { value }
+          allow(Spree::Config).to receive(name) { value }
+        end
+      end
+
+      def with_unfrozen_spree_preference_store
+        frozen_store = Spree::Config.preference_store
+        Spree::Config.preference_store = Spree::Config[:unfrozen_preference_store]
+        yield
+      ensure
+        reset_spree_preferences
```

Comment:
I'm not sure to get why this `reset_spree_preferences` is needed. If we are replacing the preference store with the frozen one generated at the test suite boot we should have it in the same state without the need of a reset, right?

Reply:
👍 Especially after my latest update, it's not needed anymore. 

One thing worth noticing that surprised me is that `reset_spree_preferences` returns configuration values to the very default ones for the preference. 

For example, `Spree::Config.mails_from` is returned to [this default value](https://github.com/solidusio/solidus/blob/3af846ba42f517c218d5ec010f42471acee2fc6b/core/lib/spree/app_configuration.rb#L242) instead of the one set in [dummy_app.rb]( https://github.com/solidusio/solidus/blob/3af846ba42f517c218d5ec010f42471acee2fc6b/core/lib/spree/testing_support/dummy_app.rb#L114). 

I think this is another good reason for avoiding to use it, or deprecate it as you already suggested.